genesyscloud/integration_credential/resource_genesyscloud_integration_credential.go

```diff
@@ -104,6 +113,15 @@ func createCredential(ctx context.Context, d *schema.ResourceData, meta interfac
 	return readCredential(ctx, d, meta)
 }
 
+func retrieveCachedOauthClientSecret(sdkConfig *platformclientv2.Configuration, fields map[string]string) {
```

Comment:
If the clientId is in the field will retrieve the secret from the cached oAuthClient